src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -110,8 +114,8 @@ private ImmutableArray<ISymbol> TryGetMemberSymbolsForLambdaParameter(IParameter
             // parameter the compiler inferred as it may have made a completely suitable inference for it.
             return parameterTypeSymbols
                 .Concat(parameter.Type)
-                .SelectMany(parameterTypeSymbol => GetMemberSymbols(parameterTypeSymbol, position, excludeInstance: false, useBaseReferenceAccessibility: false, unwrapNullable: false, isForDereference))
-                .ToImmutableArray();
+                .SelectManyAsArray(parameterTypeSymbol =>
+                    GetMemberSymbols(parameterTypeSymbol, position, excludeInstance: false, useBaseReferenceAccessibility: false, unwrapNullable, isForDereference));
```

Comment:
bug was that we were passing `unwrapNullable: false` along, instead of hte correct value that had been computed above but not passed into this helper.